### ISSUE TITLE
Global command line factor for repository rules timeout

### ISSUE THREAD

**Issue by ittaiz** [bazelbuild/bazel#6110]
### Description of the problem / feature request:

When first using Bazel in a large repo with many (thousands) of external repositories with varying types (git/maven/http) and varying sizes (KBs/Many MBs/Hundreds of MBs) one has two big problems:
1. For some reason Bazel just prints `Loading: 0 packages loaded` and seems stuck.
2. Bazel tries to download many dependencies concurrently and sometimes times out (due to contention?).

This onboarding experience is very bad and people sometimes kill Bazel since they're not sure what's going on.

### Feature requests: what underlying problem are you trying to solve with this feature?

w.r.t. problem 2 above I think a flag that adds a factor to the default timeout of each rule can be beneficial so that the first build succeeds even if it takes longer. For example if a repository rule has a timeout of 5 minutes and the cmd-line flag says 1.5 then the timeout of that repository rule will be 7.5 minutes.

### What's the output of `bazel info release`?

0.15.2

**Comment by edbaunton** [comment]
A progress indicator might also be useful so that you can know that the download/clone is actually progressing rather than just hung on e.g. a proxy. That's a separate issue but related to 1.

**Comment by aehlig** [comment]
Suggested fix: https://bazel-review.googlesource.com/c/bazel/+/75350

**Comment by aehlig** [comment]
@edbaunton 

> A progress indicator might also be useful [...]

A progress indicator is hard to implement given the current implementation of external repositories as Skyfunctions; as they are, in principle, restartable (even though we got rid of most of the restarts) all output is cached and only forwarded once the rule is finished. If we allow progress indicators that bypass this caching, we allow for situations that progress is reported and then we have to say "sorry, restarted, back to square one".  This can also be a confusing user experience.
